bin/gssstdio.c

```diff
@@ -244,8 +244,7 @@ gstd_accept(int fd, char **display_creds, char **export_name, char **mech)
 
 
 void *
-gstd_initiate(const char *hostname, const char *service, const char *princ,
-	      int fd)
+gstd_initiate(const char *hostname, const char *service, char *princ, int fd)
```

Comment:
There's a better way.  We really are treating this as const, so we should declare it as const and below, we can unconst:

```
-		name.value = (char *) princ;
+		name.value = (void *)(uintptr_t)princ;
```

In Heimdal, for example, we have a `rk_UNCONST()` macro that does exactly this.

Reply:
So do keep the const in the prototype.